### ISSUE TITLE
Reactivity rule shouldn't warn when using `props.initial*` to initialize something else

### ISSUE THREAD

**Issue by joshwilsonvu** [joshwilsonvu/eslint-plugin-solid#12]
**Describe the bug**

It's a somewhat common pattern in React and Solid to initialize a signal/state with the value of a `prop`. Given that `props` change over time, making this normally unsafe, a pseudo-standard has emerged to communicate that certain `props` are intended to be used for this purpose: prefixing them with "initial".

The `reactivity` rule should catch these cases and suppress the usual "The reactive variable 'props' should be used in a tracked scope" message. 

**To Reproduce**
To reproduce the behavior:

```jsx
function Component(props) {
  const [count, setCount] = useSignal(props.initialCount);
  return <div>{count()}</div>;
}
```

**Expected behavior**
The above snippet shouldn't warn.

**Desktop (please complete the following information):**
 - OS: Mac OS 11
 - Output of `node --version`: v16.13.0
 - Output of `npm list eslint-plugin-solid`/`yarn why eslint-plugin-solid`: v0.4.4
 - Output of `npm list eslint`/`yarn why eslint`: eslint@8.4.1
